ambry-router/src/main/java/com.github.ambry.router/NonBlockingRouter.java

```diff
@@ -469,6 +475,22 @@ int getBackgroundOperationsCount() {
     }
   }
 
+  /**
+   * Returns {@code true} if the input blobId may refer to a metadata blob, otherwise {@code false}.
+   * The method will return {@code false} only for V5 and later blobs where the BlobDataType is available.
+   * @param blobId the blobId to check.
+   * @return boolean indicating whether the blob may be metadata.
+   */
+  private static final boolean isMaybeMetadataBlob(String blobId) {
+    try {
+      BlobId.BlobDataType dataType = BlobId.getBlobDataType(blobId);
+      return (dataType == null || dataType == BlobId.BlobDataType.METADATA);
+    } catch (Exception ex) {
```

Comment:
Limit the exception to `IOException` only?

Reply:
Okay